sequence_task_1/student/include/impl.h

```diff
@@ -31,41 +31,41 @@ class StaticPlaylist
     using const_reference = typename Container::const_reference;
 
     /** @todo Iterators */
-    const_iterator begin()
+    const_iterator begin() const
     {
         return m_tracklist.cbegin();
     }
 
-    const_iterator end()
+    const_iterator end() const
     {
         return m_tracklist.cend();
     }
 
     /** @todo Constructor from any reversible sequence container */
     template<class OtherContainer>
     StaticPlaylist(const OtherContainer& otherConatiner)
-        : m_tracklist(otherConatiner.cbegin(), otherConatiner.cend())
+        : m_tracklist(otherConatiner.crbegin(), otherConatiner.crend())
     {}
 
     /** @todo Assignment from any reversible sequence container */
     template<class OtherContainer>
     Container& operator=(const OtherContainer& otherConatiner)
     {
-        m_tracklist.assign(otherConatiner.cbegin(), otherConatiner.cend());
+        m_tracklist.assign(otherConatiner.crbegin(), otherConatiner.crend());
         return m_tracklist;
     }
 
     /** @todo Add track from initializer */
     template<class... Args>
     const Song& play(Args&&... songData)
     {
-        return *m_tracklist.insert(end(), {Song(std::forward<Args>(songData))...});
+        return *m_tracklist.emplace(begin(), Song(std::forward<Args>(songData))...);
```

Comment:
Still temporary 'Song' object :(
also, why emplace is at begin? still not effective for vector

Reply:
Not clear, as we have to insert tracks to the other end of the tracks queue.
Also, the Song constructor is marked as explicit so I need to call it anyway.

Reply:
Please verify with https://github.com/smay1613/LTC-STL-Student/pull/111/files